app/models/throttle.rb

```diff
@@ -96,44 +96,138 @@ def self.max_attempts(throttle_type)
   # @return [Integer]
   def increment
     return attempts if maxed?
-    update(attempts: attempts + 1, attempted_at: Time.zone.now)
+    update(attempts: self.read_attribute(:attempts) + 1, attempted_at: Time.zone.now)
+    increment_redis
     attempts
   end
 
+  def attempts
+    if IdentityConfig.store.redis_throttle_enabled
+      @redis_attempts.to_i
+    else
+      self.read_attribute(:attempts)
+    end
+  end
```

Comment:
how would you feel about implementing a separate class? A few things:
- I don't like it when AR models talk to Redis... mixing data stores
- If our goal is to drop this table, we can simplify that future PR by leaving the AR model as-is and making a new wrapper class/service around this

Reply:
Totally understand.  I moved it into a separate service in dc8fe47.  I wasn't sure if it would be preferable to rename all of the `Throttle` uses around the application, or if it would be better for `RedisThrottle` to be absorbed into `Throttle` and then moved out of `app/models` when it's no longer interacting with the database.

I'd also like to drop the `Throttle.for` method once the database is gone, so maybe the answer is to rename everything.

Reply:
I'm a fan of "rename it now so the remove PR is simpler"

Reply:
Sounds good 👍🏼

Reply:
Rename done, though I don't love where I ended up with the unholy combination of Redis/Postgres in the throttle

Reply:
Another idea:
- Rename `Throttle` class to `DatabaseThrottle` (and use `self.table_name = :throttles` to keep it working with the same)
- Bring back `Throttle.for` ... and inside it routes between the two implementations (ex two classes with compatible APIs... maybe rename `increment` to `increment!` in both?
- `RedisThrottle` no longer needs to know about the database

Reply:
I did want to use the `Throttle` name, renaming makes sense.  The overarching service still has to interact with both Redis/Postgres to some degree since we need to write to both for backwards compatibility.

I am leaning towards limiting the changes to a rename, even if the current combination class is a bit untidy (since it should be temporary-ish).  Does that sound alright?

Reply:
Yeah that makes sense for the smoothest transition of keeping the throttle data and not doing a full reset